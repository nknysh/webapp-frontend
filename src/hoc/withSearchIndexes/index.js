export { default as withSearchIndexes } from './withSearchIndexes';
