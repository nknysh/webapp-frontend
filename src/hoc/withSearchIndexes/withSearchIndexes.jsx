import React from 'react';
import { propOr, compose } from 'ramda';
import connect from './withSearchIndexes.state';

const withSearchIndexes = ({ indexes = [] }) => WrappedComponent => {
  const ConnectedWrappedComponent = compose(connect)(WrappedComponent);

  const wrappedDisplayName = propOr(propOr('Component', 'name', WrappedComponent), 'displayName', WrappedComponent);

  const WithSearchIndexes = props => (
    <ConnectedWrappedComponent indexes={propOr(indexes, 'indexes', props)} {...props} />
  );
  WithSearchIndexes.displayName = `withSearchIndexes(${wrappedDisplayName})`;

  return WithSearchIndexes;
};

export default withSearchIndexes;
