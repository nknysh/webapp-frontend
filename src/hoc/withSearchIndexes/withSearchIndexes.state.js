import { connect } from 'react-redux';
import { reduce, merge, keys, values } from 'ramda';

import { getSearchIndex } from 'store/modules/search/selectors';

const mapStateToProps = (state, { indexes }) => {
  const buildIndex = (accum, index) => merge(accum, { [`${index}Index`]: getSearchIndex(state, index) });
  const buildIndexes = reduce(buildIndex, {});

  const searchIndexes = buildIndexes(indexes);

  return {
    indexKeys: keys(searchIndexes),
    indexes: values(searchIndexes),
  };
};

export default connect(mapStateToProps);
