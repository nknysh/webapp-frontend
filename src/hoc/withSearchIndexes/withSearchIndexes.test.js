import React from 'react';
import { compose } from 'ramda';

import withSearchIndexes from './withSearchIndexes';

const MockComponent = () => <div />;

describe('withSearchIndexes HOC', () => {
  describe('render', () => {
    it('connected component', () => {
      const Component = compose(withSearchIndexes([]))(MockComponent);
      const mountedComponent = shallow(<Component />);
      expect(mountedComponent).toMatchSnapshot();
    });
  });
});
