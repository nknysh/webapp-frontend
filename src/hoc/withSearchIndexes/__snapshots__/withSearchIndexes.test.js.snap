// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withSearchIndexes HOC render connected component 1`] = `
<Connect(MockComponent)
  indexes={Array []}
/>
`;
